### DISCUSSION TITLE
How to Handling error on multiple Database,

### DISCUSSION THREAD

**Discussion by vatoer** [prisma/prisma#23379] [Q&A] [answered]
### Question

I have 2(two) database `db1` and `db2`

and I have user in db1 

and then I'm doing error handling 

```ts
import { Prisma, User } from "@/prisma/db1/generated/client";

.....
try {
    const user = await db1.user.create({
      data: {
        email: data.email,
        name: data.name,
        password,
      },
    });
} catch (error) {
   //    console.log("[error]", Object.prototype.toString.call(error));
    if (error instanceof Prisma.PrismaClientKnownRequestError) {
      console.log(error);
      return { error: "Email already exists" };
    }
     return { error: "Unknown error" };
}
```

it always return `{ error: "Unknown error" };`  

even if the output of ` console.log("[error]", Object.prototype.toString.call(error));` said `PrismaClientKnownRequestError`

### How to reproduce (optional)

create two connection 

```tree
- prisma
  - db1
    - schema.prisma
    - generated
  - db2
    - schema.prisma
    - generated

```

### Expected behavior (optional)

if I insert to the table and let say that already have same email 

it should return the error I defined in If section 
```ts
if (error instanceof Prisma.PrismaClientKnownRequestError) {
      console.log(error);
      return { error: "Email already exists" };
    }
```

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @vatoer 👋 

Thank you for raising this question.

One thing you could try is to log the `error.code` property. According to the [documentation](https://www.prisma.io/docs/orm/prisma-client/debugging-and-troubleshooting/handling-exceptions-and-errors), each PrismaClientKnownRequestError has a code property that you can check to handle specific types of errors. For example, a unique constraint violation has the code `P2002`.
Here's how you could modify your error handling code:
```js
try {
    const user = await dbAuth.user.create({
      data: {
        email: data.email,
        name: data.name,
        password,
      },
    });
} catch (error) {
    if (error instanceof Prisma.PrismaClientKnownRequestError) {
      console.log(error);
      if (error.code === 'P2002') {
        return { error: "Email already exists" };
      }
    }
    return { error: "Unknown error" };
}
```
This way, you're specifically checking if the error code is 'P2002', which represents a unique constraint violation before returning the "Email already exists" error message. If the error is a `PrismaClientKnownRequestError` but the code is not 'P2002', it will fall through to the "Unknown error" case.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by vatoer** [reply]
the problem is that the code never reach if block 
```ts
if (error instanceof Prisma.PrismaClientKnownRequestError) {
   // never reach here
      console.log(error);
      return { error: "Email already exists" };
    }
```

your solution is possible if the code reach `if` block

**Comment by vatoer** [comment, marked as answer]
solved

the problem are in my 2 db definitions because i declare same variable `prisma` , after  I update with different `var`, problem solved

here 
```git
import { PrismaClient } from "@/prisma/db1/generated/client";

declare global {
-  var prisma: PrismaClient | undefined;
+ var prismaDb1: PrismaClient | undefined;
}

- export const db1= global.prisma|| new PrismaClient();
+ export const db1= global.prismaDb1|| new PrismaClient();

- if (process.env.NODE_ENV !== "production") global.prisma= db1;
+ if (process.env.NODE_ENV !== "production") global.prismaDb1= db1;
```

and here

```git
import { PrismaClient } from "@/prisma/db2/generated/client";

declare global {
-  var prisma: PrismaClient | undefined;
+ var prismaDb2: PrismaClient | undefined;
}

- export const db2= global.prisma|| new PrismaClient();
+ export const db2= global.prismaDb2|| new PrismaClient();

- if (process.env.NODE_ENV !== "production") global.prisma= db2;
+ if (process.env.NODE_ENV !== "production") global.prismaDb2= db2;
```
